src/x/net/http/errors.go

```diff
@@ -93,6 +102,8 @@ func WriteError(w http.ResponseWriter, err error, opts ...WriteErrorOption) {
 		fn(&o)
 	}
 
+	err = _errorRewriteFn(err)
```

Comment:
Need to hold the read lock to get the error function without causing a race.

e.g.
```
_errorRewriteFnLock.RLock()
err = _errorRewriteFn(err)
_errorRewriteFnLock.RUnlock()
```